### PR TITLE
Added origin_path option

### PR DESCRIPTION
**Pull request by sharmajee1**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [ ] All Unit Tests pass
- [ ] `rake lint` passes
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by Nirbhay1997 — COMMENTED** [review]
Please add test cases to support your changes.

**Review by Nirbhay1997 — COMMENTED** [review]
The build is failing due to some lint issue, Please run 'rubocop -a'  to rectify.
Also, use the '-s' option when doing a commit.

File: libraries/aws_cloudfront_distribution.rb [hunk 1]
```diff
@@ -44,6 +44,9 @@ def initialize(opts = {})
     # The Viewer Certificate minimum SSL version at the time of writing is TLSv1 for a default cloud certificate,
     # and for custom certifcates one of TLSv1.2_2021, TLSv1.2_2019, TLSv1.2_2018, TLSv1.1_2016, TLSv1_2016 or TLSv1
     @viewer_certificate_minimum_ssl_protocol = config.viewer_certificate[:minimum_protocol_version]
+    
+    # get origin path
+    @s3_origin_path = [config.origins.items.origin_path]
```
Comment (soumyo13) [inline libraries/aws_cloudfront_distribution.rb:49]: items is an array. So I think it should be "config.origins.items.first.origin_path"

Or you have to add a method to support this.

Comment (soumyo13) [reply libraries/aws_cloudfront_distribution.rb:49]: https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/CloudFront/Client.html#get_distribution-instance_method

**Review by soumyo13 — CHANGES_REQUESTED** [review]
Please look into my comment.

File: libraries/aws_cloudfront_distribution.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_cloudfront_distribution.rb:49]: We are getting the exception "undefined method `origin_path' for #<Array:0x00007fafd1ef6ae0>" when we are trying to run a inspec test.

describe aws_cloudfront_distribution('cloudfront-1'') do
  it { should exist }
end

Comment (sharmajee1) [reply libraries/aws_cloudfront_distribution.rb:49]: checking

Comment (soumyo13) [reply libraries/aws_cloudfront_distribution.rb:49]: There is another approach:

@distribution_config_origins = config.origins.to_h
create_resource_methods(@distribution_config_origins)

and below the initialize block add another function:
def origin_path
    items.map(&:origin_path)
end

In the inspec tests we can directly use the origin_path'
describe aws_cloudfront_distribution('sample') do
  its('origin_path') { should eq 'sample_path' }
end

**Comment by sharmajee1** [comment]
this PR became mass , creating a fresh one ,

**Pull request closed without merging**